src/coreComponents/fileIO/logPart/LogPart.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file LogPart.cpp
+ */
+
+#include "LogPart.hpp"
+#include <algorithm>
+
+namespace geos
+{
+
+LogPart::LogPart( string_view sectionTitle ):
+  m_sectionTitle( string( sectionTitle ) ),
+  m_sectionWidth( sectionTitle.length() )
+{
+  m_footerTitle = GEOS_FMT( "End : {}", m_sectionTitle );
+}
+
+void LogPart::addDescription( string_view description )
+{
+  m_descriptions.push_back( string( description ) );
+}
+
+void LogPart::addEndDescription( string_view description )
+{
+  m_endLogMessages.push_back( string( description ) );
+}
+
+void LogPart::setMinWidth( integer const & minWidth )
+{
+  m_rowMinWidth = minWidth;
+}
+
+void LogPart::formatAndInsertDescriptions( std::vector< string > & descriptionContainer,
+                                           string_view name,
+                                           std::vector< string > const & values )
+{
+  string const nameFormatted =  GEOS_FMT( "- {}: ", string( name ));
```

Comment:
I think that it is the responsability of the class user to add those `"- "` at the start of their texts (imagine if you only have one description).

Reply:
if we have one description we will not have the `"- "`, it's only for mutiple values in the description